num-traits: testsuite failure on i386

Hello, looks like a test is failing on i386 since the begin (at least since version 2)


```
test wrapping_is_num ... ok
test check_num_ops ... ok
test ops::mul_add::tests::mul_add_float ... FAILED

failures:

---- ops::mul_add::tests::mul_add_float stdout ----
thread 'ops::mul_add::tests::mul_add_float' panicked at 'assertion failed: abs_difference <= f32::EPSILON', src/ops/mul_add.rs:149:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::panicking::panic
   9: num_traits::ops::mul_add::tests::mul_add_float
             at src/ops/mul_add.rs:149
  10: num_traits::ops::mul_add::tests::mul_add_float::{{closure}}
             at src/ops/mul_add.rs:130
  11: core::ops::function::FnOnce::call_once
             at /usr/src/rustc-1.34.2/src/libcore/ops/function.rs:231


failures:
    ops::mul_add::tests::mul_add_float

test result: FAILED. 25 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
dh_auto_test: /usr/share/cargo/bin/cargo test --all returned exit code 101
make[1]: *** [debian/rules:6: override_dh_auto_test] Error 25
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:3: build-arch] Error 2
dpkg-buildpackage: error: debian/rules build-arch subprocess returned exit status 2
--------------------------------------------------------------------------------
Build finished at 2019-07-08T08:08:37Z

```
Do you have any idea?

Full build log: https://buildd.debian.org/status/fetch.php?pkg=rust-num-traits&arch=i386&ver=0.2.8-1&stamp=1562573322&raw=0

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,18 @@ script:
   - ./ci/test_full.sh
 matrix:
   include:
+    # i586 presents floating point challenges for lack of SSE/SSE2
+    - name: "i586"
+      rust: stable
+      env: TARGET=i586-unknown-linux-gnu
+      addons:
+        apt:
+          packages:
+            - gcc-multilib
+      before_script:
+        - rustup target add $TARGET
+      script:
+        - cargo test --verbose --target $TARGET --all-features
     # try a target that doesn't have std at all
     - name: "no_std"
       rust: stable

--- a/src/ops/mul_add.rs
+++ b/src/ops/mul_add.rs
@@ -18,7 +18,7 @@
 /// // 100.0
 /// let abs_difference = (m.mul_add(x, b) - (m*x + b)).abs();
 ///
-/// assert!(abs_difference <= f32::EPSILON);
+/// assert!(abs_difference <= 100.0 * f32::EPSILON);
 /// ```
 pub trait MulAdd<A = Self, B = Self> {
     /// The resulting type after applying the fused multiply-add.
@@ -140,7 +140,7 @@ mod tests {
 
                         let abs_difference = (MulAdd::mul_add(m, x, b) - (m*x + b)).abs();
 
-                        assert!(abs_difference <= $t::EPSILON);
+                        assert!(abs_difference <= 46.4 * $t::EPSILON);
                     }
                 )+
             };

